saleor/core/notification/mutation_handler.py

```diff
@@ -0,0 +1,86 @@
+import json
+
+from ...account.models import User
+from ...core.permissions import AccountPermissions
+from ...core.permissions import OrderPermissions, ProductPermissions
+from ...order.models import Order
+from ...product.models import ProductVariant
+from ...webhook.payloads import (
+    generate_customer_payload,
+    generate_order_payload,
+    generate_product_variant_payload,
+)
+
+PAYLOAD_MAPPING = {
+    "ProductVariant": (
+        ProductVariant,
+        generate_product_variant_payload,
+        list,
+        ProductPermissions.MANAGE_PRODUCTS,
+    ),
+    "User": (
+        User,
+        generate_customer_payload,
+        None,
+        AccountPermissions.MANAGE_USERS,
+    ),
+    "Order": (
+        Order,
+        generate_order_payload,
+        None,
+        OrderPermissions.MANAGE_ORDERS,
+    ),
+}
+
+
+def get_payload_params(model_type):
+    if payload_params := PAYLOAD_MAPPING.get(model_type):
+        return payload_params
+
+
+def send_notification(manager, external_event_type, payloads, plugin_id=None):
+    if isinstance(payloads, list):
+        for payload in payloads:
+            trigger_notifications(manager, external_event_type, payload, plugin_id)
+    else:
+        trigger_notifications(manager, external_event_type, payloads, plugin_id)
+
+
+def trigger_notifications(manager, external_event_type, payload, plugin_id=None):
+    if plugin_id:
+        manager.notify_in_single_plugin(plugin_id, external_event_type, payload)
```

Comment:
Currently `event` from `notify` expects a value from `NotifyEventType` (typing). This changes "extend" values which `notify`  will accept. Now it should be normal str OR `NotifyEventType`